src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -615,32 +578,96 @@ static void checkApiRateLimit(@NonNull TaskListener listener, GitHub github)
 
         @Override
         public long getRecurrencePeriod() {
-            return TimeUnit.SECONDS.toMillis(15);
+            return TimeUnit.MINUTES.toMillis(2);
         }
 
         @Override
         protected void doRun() throws Exception {
             // free any connection unused for the last 5 minutes
             long threshold = System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(5);
-            synchronized (githubs) {
-                for (Iterator<Map.Entry<GitHub, Long>> iterator = lastUsed.entrySet().iterator();
-                     iterator.hasNext(); ) {
-                    Map.Entry<GitHub, Long> entry = iterator.next();
-                    Long lastUse = entry.getValue();
-                    if (lastUse == null || lastUse < threshold) {
-                        iterator.remove();
-                        unused(entry.getKey());
+            synchronized (connections) {
+                GitHubConnection.removeAllUnused(threshold);
+            }
+        }
+    }
+
+    private static class GitHubConnection {
+        @NonNull
+        private final GitHub gitHub;
+
+        @CheckForNull
+        private final Cache cache;
+
+        private final boolean cleanupCacheFolder;
+        private int usageCount = 1;
+        private Long lastUsed = System.currentTimeMillis();
```

Comment:
No need to use a boxed `Long` here. It adds overhead, and makes it possible for someone to accidentally introduce NPEs in the future since the current code assumes it is non-null.

```suggestion
        private long lastUsed = System.currentTimeMillis();
```